packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,480 @@
+ * 1. when the body and content type are null, we return an empty openapi body
+ * 2. when the content type is an invalid one, we return INVALID_CONTENT_TYPE error
+ * 3. when the content type is json based content types, we convert that to openapi properties recursively
+ * 4. multipart/formdata is also handled differently
+ * 5. in other valid content types, the body is handled as a string
+ */
+export const generateOpenApiRequestBody = (
+  hoppRequestBody: HoppRESTReqBody
+): E.Either<RequestBodyGenerationError, OpenAPIV3.RequestBodyObject> =>
+  pipe(
+    hoppRequestBody,
+    E.fromPredicate(isEmptyBody, () => "NON_EMPTY_BODY" as const),
+    E.map(generateOpenAPIBodyForEmptyBody),
+    E.altW(() =>
+      pipe(
+        hoppRequestBody.contentType,
+        E.fromPredicate(
+          isValidContentType,
+          () => "INVALID_CONTENT_TYPE" as const
+        ),
+        E.chainW((contentType) =>
+          pipe(
+            hoppRequestBody,
+            O.fromPredicate(isNonFormDataBody),
+            O.chain((hoppRequestBody) =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(() => isContentTypeJSON(contentType)),
+                O.chain(flow(({ body }) => safeParseJSON(body))),
+                O.map(generateOpenAPIBodyForJSON(contentType)),
+                O.alt(() =>
+                  pipe(
+                    hoppRequestBody,
+                    O.fromPredicate(isContentTypeUrlEncoded),
+                    O.map(({ body }) => generateOpenAPIBodyForUrlEncoded(body)),
+                    O.alt(() =>
+                      pipe(
+                        hoppRequestBody,
+                        generateOpenAPIBodyForPlainText,
+                        O.some
+                      )
+                    )
+                  )
+                )
+              )
+            ),
+            O.alt(() =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(isFormDataBody),
+                O.map(generateOpenAPIBodyForFormData)
+              )
+            ),
+            E.fromOption(() => "INVALID_BODY" as const)
+          )
+        )
+      )
+    )
+  )
+
+const OpenAPIAuthNames = {
+  basic: "basicAuth",
+  "api-key": "ApiKeyAuth",
+  bearer: "bearerAuth",
+} as const
+
+export const generateOpenApiAuth = (
+  hoppAuth: HoppRESTAuth
+): E.Either<"INVALID_AUTH", OpenAPIV3.SecurityRequirementObject> =>
+  pipe(
+    hoppAuth.authType,
+    O.fromPredicate((authType) => authType === "none"),
+    O.map(() => ({})),
+    O.alt(() =>
+      pipe(
+        hoppAuth.authType,
+        O.fromPredicate(
+          (authType): authType is "basic" | "api-key" | "bearer" =>
+            ["basic", "api-key", "bearer"].includes(authType)
+        ),
+        O.map((authType) =>
+          pipe(OpenAPIAuthNames, (auths) => ({
+            [auths[authType]]: [],
+          }))
+        )
+      )
+    ),
+    E.fromOption(() => "INVALID_AUTH" as const)
+  )
```

Comment:
More cleaner implementation ?

```suggestion
  hoppAuth.authType === "none"
    ? E.right({})
    : pipe(
        hoppAuth.authType,
        O.fromPredicate(
          (authType): authType is "basic" | "api-key" | "bearer" =>
            ["basic", "api-key", "bearer"].includes(authType)
        ),
        O.map((authType) =>
          pipe(OpenAPIAuthNames, (auths) => ({
            [auths[authType]]: [],
          }))
        ),
        E.fromOption(() => "INVALID_AUTH" as const)
      )
```